### PR TITLE
[CFM-861-fix] Activity stream items view, for restricted groups.

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-861
https://issuetracker.amplexor.com/jira/browse/CFM-1194

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 1]
```diff
@@ -3587,16 +3587,30 @@ function c4m_message_query_activity_stream_entity_field_access_alter(
       $subquery = db_select('node_access', 'na')
         ->fields('na', array('nid'));
 
+      $subquery
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: COMMENTS

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 2]
```diff
@@ -3587,16 +3587,34 @@ function c4m_message_query_activity_stream_entity_field_access_alter(
...
-          );
+          $grant_condition = db_and()
+            ->condition('na.gid', $gid)
+            ->condition('na.realm', $realm);
+
+          // For pluggable node access grants, add condition on 'c4m_og_status'
+          // field - set allowed values to published and archived.
+          if (in_array($realm, array_keys($pluggable_node_access_grants))) {
+            $grant_condition->condition(
+              db_or()
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: don't use shortcuts ....
take it out to a variable ...

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 3]
```diff
@@ -3587,16 +3587,34 @@ function c4m_message_query_activity_stream_entity_field_access_alter(
...
-          $grant_conditions->condition(
-            db_and()
-              ->condition('na.gid', $gid)
-              ->condition('na.realm', $realm)
-          );
+          $grant_condition = db_and()
+            ->condition('na.gid', $gid)
+            ->condition('na.realm', $realm);
+
+          // For pluggable node access grants, add condition on 'c4m_og_status'
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: comment is unclear

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 4]
```diff
@@ -3587,16 +3587,34 @@ function c4m_message_query_activity_stream_entity_field_access_alter(
       $subquery = db_select('node_access', 'na')
         ->fields('na', array('nid'));
 
+      // Pluggable node access grunts will require extra condition, based on
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: comment is unclear

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 5]
```diff
@@ -3587,16 +3587,34 @@ function c4m_message_query_activity_stream_entity_field_access_alter(
...
+      // group state field. Hence, need to join, to resolve 'c4m_og_status'
+      // field.
+      $subquery
+        ->leftJoin('field_data_c4m_og_status', 'gst', 'na.nid = gst.entity_id');
+      // Store pluggable node access grunts to identify them latter.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: comment is unclear

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 6]
```diff
@@ -3587,16 +3587,34 @@ function c4m_message_query_activity_stream_entity_field_access_alter(
       $subquery = db_select('node_access', 'na')
         ->fields('na', array('nid'));
 
+      // Pluggable node access grunts will require extra condition, based on
+      // group state field. Hence, need to join, to resolve 'c4m_og_status'
+      // field.
+      $subquery
+        ->leftJoin('field_data_c4m_og_status', 'gst', 'na.nid = gst.entity_id');
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: should be under one line.

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 7]
```diff
@@ -3587,16 +3587,36 @@ function c4m_message_query_activity_stream_entity_field_access_alter(
...
-              ->condition('na.realm', $realm)
-          );
+          $grant_condition = db_and()
+            ->condition('na.gid', $gid)
+            ->condition('na.realm', $realm);
+
+          // If grants is allocated by pluggable_node_access model,
+          // add group status condition - status may be either published
+          // or archived.
+          if (in_array($realm, array_keys($pluggable_node_access_grants))) {
```
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: Private and public groups do not have grants from pluggable_node_access module.

**Comment by anvmn** [comment]
out

**Pull request merged**